File: src/main/java/org/aldo/api/service/interfaces/ProjectsService.java
package org.aldo.api.service.interfaces;

import org.aldo.api.data.dto.ProjectDto;
import org.aldo.api.data.dto.SummaryProjectDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface ProjectsService {
    Page<SummaryProjectDto> getAllProjectsCriteria(String direction, String criteria, Integer duration, Boolean pnrr, String name, Pageable pageable);
    List<ProjectDto> getAllProjects();
}

File: src/app/service/projects/projects.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Project} from "../../model/Project";
import {SummaryProjectDto} from "../../model/dto/SummaryProjectDto";
import {Page} from "../../model/Page";
import {UpdateProjectDto} from "../../model/dto/UpdateProjectDto";

@Injectable({
  providedIn: 'root'
})
export class ProjectsService {
  private urlApi = 'http://localhost:8080/api/v1/';

  constructor(private http: HttpClient) { }

  getProject(id: number) {
    return this.http.get<Project>(this.urlApi + `project/${id}`)
  }

  getProjects(direction: string, criteria: string, name: string, duration: number, pnrr: boolean | string) {
    return this.http.get<Page<SummaryProjectDto>>(
      this.urlApi + 'projects',
      {params: {direction, criteria, duration: duration.toString(), pnrr: pnrr.toString(), name}})
  }

  addProject(project: Project) {
    return this.http.post(this.urlApi + 'project', project)
  }

  updateProject(project: UpdateProjectDto) {
    return this.http.patch(this.urlApi + 'project', project)
  }
}
